### PR TITLE
Issue #2709: Consider replacing the home button with a Back to site b…

### PR DESCRIPTION
**Pull request by alexfinnarn**
…utton

### CODE REVIEW

File: core/modules/admin_bar/admin_bar.inc [hunk 1]
```diff
@@ -704,19 +704,28 @@ function admin_bar_links_alert() {
...
     '#type' => 'checkbox',
     '#title' => t('Keep bar at top of page'),
     '#default_value' => $config->get('position_fixed'),
     '#description' => t('Displays the administration bar always at the top of the browser viewport, even when scrolling the page.'),
   );
 
+  $form['back_to_site'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Add Back To Site Link'),
+    '#default_value' => $config->get('back_to_site'),
```
Comment (opi) [inline core/modules/admin_bar/admin_bar.inc:725]: I like when variable name are almost self-explanatory, what about something like `backtosite_link` ?

File: core/modules/admin_bar/admin_bar.module [hunk 2]
```diff
@@ -243,6 +243,15 @@ function admin_bar_preprocess_page(&$variables) {
     $settings['position_fixed'] = $setting;
   }
 
+  // Send whether this is an admin page to JS for back to site functionality.
+  $settings['current_path_is_admin'] = false;
```
Comment (opi) [inline core/modules/admin_bar/admin_bar.module:247]: use uppercased `TRUE` and `FALSE` constants.

File: core/modules/admin_bar/admin_bar.module [hunk 3]
```diff
@@ -243,6 +243,15 @@ function admin_bar_preprocess_page(&$variables) {
...
+  $settings['current_path'] = current_path();
+  if (path_is_admin($settings['current_path'])) {
+    $settings['current_path_is_admin'] = true;
```
Comment (opi) [inline core/modules/admin_bar/admin_bar.module:250]: could be one-lined: 
`$settings['current_path_is_admin'] = path_is_admin($settings['current_path'])`.

File: core/modules/admin_bar/js/admin_bar.js [hunk 4]
```diff
@@ -41,6 +41,8 @@ Backdrop.behaviors.adminBar = {
           setTimeout(function () {
             $adminBar.css({visibility: ''});
           }, 0);
+
```
Comment (opi) [inline core/modules/admin_bar/js/admin_bar.js:44]: unnecessary line breaks

File: core/modules/admin_bar/admin_bar.inc [hunk 5]
```diff
@@ -704,19 +704,28 @@ function admin_bar_links_alert() {
...
   $form['position_fixed'] = array(
     '#type' => 'checkbox',
     '#title' => t('Keep bar at top of page'),
     '#default_value' => $config->get('position_fixed'),
     '#description' => t('Displays the administration bar always at the top of the browser viewport, even when scrolling the page.'),
   );
 
+  $form['back_to_site'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Add Back To Site Link'),
```
Comment (opi) [inline core/modules/admin_bar/admin_bar.inc:724]: should be lowercased, as @BWPanda said in his [review](https://github.com/backdrop/backdrop-issues/issues/2709#issuecomment-390500108)

**Review by opi — CHANGES_REQUESTED** [review]
Thanks for working on this nice UX improvment ! I noticed few stuff to fix. I'm ready to discuss and help.

**Comment by quicksketch** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/4236. Thanks @alexfinnarn!

**Pull request closed without merging**